cmd/rb-main.go

```diff
@@ -0,0 +1,207 @@
+	contentCh := make(chan *clientContent)
+	errorCh := clnt.Remove(isIncomplete, isRemoveBucket, contentCh)
+
+	for content := range clnt.List(true, false, DirLast) {
+		if content.Err != nil {
+			errorIf(content.Err.Trace(url), "Failed to remove `"+url+"`.")
+			switch content.Err.ToGoError().(type) {
+			case PathInsufficientPermission:
+				// Ignore Permission error.
+				continue
+			}
+			close(contentCh)
+			return exitStatus(globalErrorExitStatus)
+		}
+		urlString := content.URL.Path
+
+		sent := false
+		for !sent {
+			select {
+			case contentCh <- content:
+				sent = true
+			case pErr := <-errorCh:
+				errorIf(pErr.Trace(urlString), "Failed to remove `"+urlString+"`.")
+				switch pErr.ToGoError().(type) {
+				case PathInsufficientPermission:
+					// Ignore Permission error.
+					continue
+				}
+				close(contentCh)
+				return exitStatus(globalErrorExitStatus)
+			}
+		}
+	}
+
+	close(contentCh)
+	for pErr := range errorCh {
+		errorIf(pErr.Trace(url), "Failed to remove `"+url+"`.")
+		switch pErr.ToGoError().(type) {
+		case PathInsufficientPermission:
+			// Ignore Permission error.
+			continue
+		}
+		return exitStatus(globalErrorExitStatus)
+	}
+	return nil
+}
+
+// isNamespaceRemoval returns true if alias
+// is not qualified by bucket
+func isNamespaceRemoval(url string) bool {
+	// clean path for aliases like s3/.
+	//Note: UNC path using / works properly in go 1.9.2 even though it breaks the UNC specification.
+	url = filepath.ToSlash(filepath.Clean(url))
+	// namespace removal applies only for non FS. So filter out if passed url represents a directory
+	if !isAliasURLDir(url, nil) {
+		_, path := url2Alias(url)
+		return (path == "")
+	}
+	return false
+}
+
+// mainRemoveBucket is entry point for rb command.
+func mainRemoveBucket(ctx *cli.Context) error {
+	// check 'rb' cli arguments.
+	checkRbSyntax(ctx)
+	// Set command flags from context.
+	isForce := ctx.Bool("force")
+	// Additional command speific theme customization.
+	console.SetColor("RemoveBucket", color.New(color.FgGreen, color.Bold))
+
+	var cErr error
+	for _, targetURL := range ctx.Args() {
+		// Instantiate client for URL.
+		clnt, err := newClient(targetURL)
+		if err != nil {
+			errorIf(err.Trace(targetURL), "Invalid target `"+targetURL+"`.")
+			cErr = exitStatus(globalErrorExitStatus)
+			continue
+		}
+		isEmpty := true
+		for range clnt.List(true, false, DirNone) {
+			isEmpty = false
+			break
+		}
+		// For all recursive operations make sure to check for 'force' flag.
+		if !isForce {
+			if isNamespaceRemoval(targetURL) {
+				fatalIf(errDummy().Trace(),
+					"This operation results in **site-wide** removal of buckets. If you are really sure, retry this command with ‘--force’ flags.")
+			}
+			if !isEmpty {
+				fatalIf(errDummy().Trace(), "`"+targetURL+"` is not empty. Retry this command with ‘--force’ flag if you want to remove `"+targetURL+"` and all its contents")
+			}
+		}
+		if e := deleteBucket(targetURL); e == nil {
```

Comment:
Why we don't print errors (`err != nil` case) ?

Reply:
@vadmeste,changed this to print error also.